common-io/tests/unity_main.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * unity_main.c
+ *
+ *  Created on: Sep 21, 2023
+ *      Author: cookpate
+ */
+
+#include "unity.h"
+#include "unity_fixture.h"
+#include "unity_internals.h"
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Target includes */
+#include "target/core-v-mcu/include/core-v-mcu-system.h"
+
+#include "iot_gpio.h"
+
+static void unity_main( void *param );
+
+int main( void )
+{
+	BaseType_t xResult;
+	TaskHandle_t pxTaskHandle =
+	{ 0 };
+
+	system_init();
+
+	xResult = xTaskCreate( &unity_main, "UnityTest", 3000U, // Task will allocate 3000 heap bytes for stack
+		NULL, 2U,	// Created task at second-lowest priority
+		&pxTaskHandle );
+
+	configASSERT( xResult == pdPASS );
+
+	/* Start the tasks and timer running. */
+	vTaskStartScheduler();
+
+	/* If all is well, the scheduler will now be running, and the following
+	 line will never be reached.  If the following line does execute, then
+	 there was insufficient FreeRTOS heap memory available for the Idle and/or
+	 timer tasks to be created.  See the memory management section on the
+	 FreeRTOS web site for more details on the FreeRTOS heap
+	 http://www.freertos.org/a00111.html. */
+
```

Comment:
should make this line an assert so the lack of memory will cause an immediate crash at a specific point.